### ISSUE TITLE
Resolve data limit on output passed in the error reporting step of SFN

### ISSUE THREAD

**Issue by joyceyan** [chanzuckerberg/single-cell#690]
When running the schema 5.0 migration, we encountered an issue with the SFN failing due to having too much output passed in the error reporting step. We mistakenly thought we resolved this by reducing the amount of information we log to make the payload smaller, and a subsequent run of the SFN (in dev) succeeded. However, upon closer inspection, it seems like by the time we get to the reporting step of the SFN, we've already successfully migrated the dataset and it won't be picked up for migration by a subsequent run of the SFN.

This is probably best resolved by uploading the error reports to somewhere in S3, instead of trying to pass the payload around within the SFN directly. This will also have the benefit of us having more control over how long the error reports are retained for.

**Comment by brianraymor** [comment]
@metakuni - moving this to chanzuckerberg/single-cell-data-portal#6790 for triage.

**Comment by danieljhegeman** [comment]
This will likely be obviated by changes to the migration logic that will include greater usage of s3 for logging instead of reporting details of errors/successes to the step function output itself. TBD based on specifics of plan that results from this [tech spec](https://docs.google.com/document/d/1r6joea1JQi2MX3R4V1Jo9Ph6QZH2YjRDgYtJbPfIUMA/edit).

**Comment by joyceyan** [comment]
@danieljhegeman Understood and agreed, as I also suggested persisting logs in S3 in the ticket body as the best solution for solving this. To clarify, the purpose of this ticket is to capture the "business problem" so that Brian and Kuni can understand how to prioritize the work. If we didn't have this problem, we wouldn't consider changing our logging strategy.

**Comment by metakuni** [comment]
Removing from 5.0 but for resolution before 5.1. If we implement the migration efficiency proposal and it's addressed by that, we can close this issue then.

**Comment by brianraymor** [comment]
@metakuni -moving this back to https://github.com/chanzuckerberg/single-cell-data-portal/issues/6790 per my [comments above](https://github.com/chanzuckerberg/single-cell/issues/690#issuecomment-2021621490).  Per the conversation today, I'd also like to keep migration issues decoupled from delivering 5.1.0.

**Comment by metakuni** [comment]
Hmm, ok. I was thinking of using #648 for the infrastructure work and creating a separate epic for the migration, but that makes no sense in hindsight. 🤦 I'll create a separate (pre-5.1) infra epic that we'll tackle before the platform work in #648.

**Comment by danieljhegeman** [comment]
I've made reference to completing this ticket in the Implementation section of the tech spec mentioned in my comment above

**Comment by danieljhegeman** [comment]
Holding off on this since we are not likely to make large-scale changes to the schema migration algorithm in the immediate future (i.e., ahead of Schema 5.1).

**Comment by danieljhegeman** [comment]
Some more investigation yields the following: this error appears to have happened primarily due to executing the SFN on a corpus that had a broken link between old and new dataset versions. This generated a ton of errors of the flavor
```
{
  "step": "dataset_migrate",
  "error": "An error occurred (404) when calling the HeadObject operation: Not Found",
  "args": [],
  "kwargs": {
    "collection_version_id": <redacted for example>,
    "collection_id": <redacted for example>,
    "dataset_id": <redacted for example>,
    "dataset_version_id": <redacted for example>
  }
}
```

**Comment by brianraymor** [comment]
> a broken link between old and new dataset versions

Could you say more about how that condition arises?

**Comment by danieljhegeman** [comment]
In the most recent invocation that resulted in this particular error, we had 510 datasets that failed. Again, largely failing with the error detailed immediately above that is an artifact of running the SFN in an adhoc fashion where the original version for a dataset could not be easily retrieved. In other words, it is _not_ because we're outputting too much data for individually-erroring datasets that we might expect during the course of a normal migration run.

This sort of thing should not be happening in the future, because it was all downstream of deploying to dev in the middle of a step function. If we don't do that, then we won't run into this error either. I recommend we not take action on this issue at this time, as it is not likely to pay dividends cc @metakuni @Bento007

**Comment by danieljhegeman** [comment]
> > a broken link between old and new dataset versions
> 
> Could you say more about how that condition arises?

@brianraymor this is the scenario that comes from a mid-sfn-run deployment which breaks the step function between the 'dataset migrate' step and the ingestion step. It causes the db to end up in a funny state where dataset version rows exist but no assets are in s3 under that version id. I ended up adding temporary adhoc code to recover some (possible for revisions only; not unpublished private) for the sake of the dev migration testing when this occurred originally.

Recommended course of action in the future is to simply reset the db and start from scratch rather than attempting to hobble through a (test) migration with these broken links.

To summarize: this will not happen as long as Engineering doesn't make mistakes by deploying mid-run again. And if they do that, they need to start over.

**Comment by danieljhegeman** [comment]
Unnecessary per reasoning given above
